packages/core/crates/core-p2p/src/api.rs

```diff
@@ -0,0 +1,206 @@
+use std::pin::Pin;
+use futures::{Stream, channel::mpsc, future::poll_fn};
+use futures_lite::StreamExt;
+use libp2p::identity::PeerId;
+
+use core_network::messaging::ControlMessage;
+use utils_log::error;
+
+use crate::errors::{Result, P2PError};
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct HeartbeatChallenge(pub PeerId, pub ControlMessage);
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct ManualPingChallenge(pub PeerId, pub ControlMessage);
+
+
+// TODO: NOTE: UnboundedSender and UnboundedReceiver are bound only by available memory
+// in case of faster input than output the memory might run out
+pub type HeartbeaRequestRx = mpsc::UnboundedReceiver<(PeerId, ControlMessage)>;
```

Comment:
How are these queues different from the ones  in `messaging.rs` ?

Reply:
This is an `mpsc`, multiple producers -> single consumer.